OpenRA.Mods.Common/SpriteLoaders/GenieSlpLoader.cs

```diff
@@ -89,11 +115,11 @@ static byte[] ReadRowCommands(Stream stream, GenieSlpFrameHeader header)
 						break;
 
 					case 0x03: // Skip Pixels (long)
-						rowPosX += GetTopNibblePlusNext(currByte);
+						rowPosX += (ushort)GetTopNibblePlusNext(currByte, stream);
 						break;
 
 					case 0x06: // Copy & Transform
-						for (uint i = 0, cmdLen = GetTopNibblePlusNext(currByte, stream); i < cmdLen; i++)
+						for (uint i = 0, cmdLen = GetTopNibbleOrNext(currByte, stream); i < cmdLen; i++)
```

Comment:
Had to change it because it crashed.
Also confirmed the change with [this](https://github.com/Phrohdoh/aoe-parser-cs/blob/master/AgeSlp/SlpFrameRow.cs#L120).